kratos/processes/calculate_nodal_area_process.cpp

```diff
@@ -0,0 +1,156 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//  Collaborators:   Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "utilities/geometry_utilities.h"
+#include "processes/calculate_nodal_area_process.h"
+
+namespace Kratos
+{
+
+template<bool THistorical>
+void CalculateNodalAreaProcess<THistorical>::Execute()
+{
+    KRATOS_TRY
+
+    // Check if variables are available
+    if (THistorical) {
+        KRATOS_ERROR_IF_NOT(mrModelPart.NodesBegin()->SolutionStepsDataHas( NODAL_AREA )) << "Variable NODAL_AREA not in the model part!" << std::endl;
```

Comment:
Why not check the ModelPart itself?
Won't this check crash if there are no nodes?

Reply:
Well, you are right, but I think if there are no nodes you should not use this